google_nest_sdm/webrtc_util.py

```diff
@@ -0,0 +1,114 @@
+
+        Returns:
+            str: The direction of the media track. One of 'sendrecv', 'sendonly', 'recvonly', or 'inactive'.
+        """
+        # Split the SDP into lines
+        lines = sdp.splitlines()
+
+        # Variable to track if we are in the desired media section
+        in_media_section = False
+
+        # Search for the media type and direction
+        for line in lines:
+            # Check if the line is a media description line (m=)
+            if line.startswith('m='):
+                in_media_section = (kind in line)
+
+            # If we're in the desired media section, check for direction
+            if in_media_section:
+                for direction in AVAILABLE_DIRECTIONS:
+                    if line.startswith(f'a={direction}'):
+                        return direction
+        return None
+
+    def update_direction_in_answer(self, answer_sdp, kind, old_direction, new_direction):
+        """
+        Updates the direction of a specific media track in the SDP answer if it matches a certain direction.
+
+        Args:
+            answer_sdp (str): The SDP answer in string format.
+            kind (str): The kind of media track to update ('audio' or 'video').
+            old_direction (str): The old direction to find ('sendrecv', 'sendonly', 'recvonly', 'inactive').
+            new_direction (str): The new direction to set ('sendrecv', 'sendonly', 'recvonly', 'inactive').
+
+        Returns:
+            str: The updated SDP with the new direction.
+        """
+
+        # Update the SDP
+        sdp_lines = answer_sdp.split('\r\n')
+        updated_sdp_lines = []
+        in_media_section = False
+
+        for line in sdp_lines:
+            if line.startswith('m='):
+                in_media_section = (kind in line)
+            if in_media_section and line.startswith('a='):
+                # Update the direction line if it matches the kind
+                if line.startswith(f'a={old_direction}'):
+                    updated_sdp_lines.append(f'a={new_direction}')
+                    continue
+            updated_sdp_lines.append(line)
+
+        updated_sdp = '\r\n'.join(updated_sdp_lines)
+        return updated_sdp
+
+    def add_foundation_to_candidates(self, sdp):
+        """
+        Adds a foundation value to all ICE candidates in the SDP if it does not already exist.
+
+        Args:
+            sdp (str): The SDP in string format.
+
+        Returns:
+            str: The updated SDP with the foundation value added to ICE candidates.
+        """
+        # Add the foundation value to each ICE candidate line
+        updated_sdp = sdp
+        index = 1
+        for line in sdp.split("\r\n"):
+            if line.startswith("a=candidate: "):
+                updated_sdp = updated_sdp.replace(line, line.replace(
+                    "a=candidate: ", "a=candidate:"+str(index)+" "))
+                index += 1
+        return updated_sdp
```

Comment:
I think this could be a little easier to reason about if the string replacement is narrowly scoped:

```suggestion
        lines = sdp.split("\r\n")
        updated_lines = []
        index = 1
        for line in lines:
            if line.startswith("a=candidate: "):
                line.replace("a=candidate: ", f"a=candidate:{index} "))
                index += 1
            updated_lines.append(lines)
        return "\r\n".join(updated_lines)
```

I noticed you also did it this way in the other function, and i like that.